src/braket/aws/aws_device.py

```diff
@@ -569,7 +569,9 @@ def get_devices(
             types (Optional[list[AwsDeviceType]]): device type list, default is `None`
                 QPUs will be searched for all regions and simulators will only be
                 searched for the region of the current session.
-            statuses (Optional[list[str]]): device status list, default is `None`
+            statuses (Optional[list[str]]): device status list, default is `None`. When `None`
+                is used, RETIRED devices will not be returned. To include RETIRED devices in
+                the results, add `RETIRED` to the list passed to this parameter.
```

Comment:
We typically state the default semantically used, even if the function signature has `=None`. Here, I believe the default is `["ONLINE", "OFFLINE"]`. We can add a note in the doc string that the default behavior is to filter out `"RETIRED"`

Reply:
I am generally not a fan of listing this stuff out here since it can more easily go out-of-date (like if we add another state). Then, for users of older versions of the BDK, it would actually be wrong as it would return more than those two.

Reply:
The docs are versioned by release, so I'm not sure I see the issue. The doc string is wrong for older versions of the bdk regardless; I think it's just a question of the phrasing here

Reply:
Oh I see what you mean by listing stuff out. Would it be better to just add a new parameter `filter_retired=True` that kicks in when `statuses` is `None` to separate the logic?

Reply:
I considered it, butI think it would be confusing to add a parameter that can have a direct opposite effect of another parameter (ie. have a filter_retired=True and statuses=["RETIRED"])

Reply:
I see that "default is `None`" is consistent with the other doc strings. I don't love how the argument doc strings don't actually specify the default behavior, just the default value of the parameter.

I'd prefer to see these called out as filters explicitly, maybe even just changing "device name list, default is `None`" to "device name filter, default is `None`". Then we could clarify statuses like "device status filter, default is `None`. When `None` is used, RETIRED devices will not be returned. To include RETIRED devices in the results, use a filter including `"RETIRED"` for this parameter."

Reply:
Good call